include/dyad/stream/dyad_stream_api.hpp

```diff
@@ -70,11 +71,17 @@ void fsync_ofstream (std::basic_ofstream<_CharT, _Traits>& os)
         os.flush ();
         fsync (static_cast<my_filebuf&> (*os.rdbuf ()).handle ());
     }
+#else
+    if (os.is_open ()) {
+        os.flush ();
```

Comment:
Why do we need to do `os.flush()`?

Reply:
To make sure that data is cleared out before starting the followup synchronization routine, which can happen before d'tor call.

Reply:
I would not do that always. As I said, we should avoid any kind of flush.

Reply:
For any fencing semantic, I think this is needed.

Reply:
the session semantics is maintained by closing already, we don't need that for most file systems.

Reply:
This is decoupled from file close. This is a utility function like mem fence instruction. If you use it just before stream close, there will be nothing to flush at close.

Reply:
So this wont be called automatically by close of stream? Users need to call it explicitly?

Reply:
My understanding is that this would not be an expensive operation. Therefore its ok.